inject/cdi2-se/src/main/java/org/glassfish/jersey/inject/cdi/se/bean/SupplierBeanBridge.java

```diff
@@ -143,4 +143,9 @@ public void destroy(Object instance, CreationalContext context) {
     public Class<? extends Annotation> getScope() {
         return binding.getScope() == null ? Dependent.class : transformScope(binding.getScope());
     }
+
+    @Override
+    public Class<?> getBeanClass() {
```

Comment:
if (binding.getContracts().size() == 1) {
    return binding.getContracts().iterator().next();
} else {
   return super.getBeanClass();
}

Reply:
OK, but how is this different than SupplierThreadScopeBean that has the same expression, i.e.

    return (Class<?>) this.binding.getContracts().iterator().next()

Reply:
Right, but better safe than NPE